File: AN_2/SEM_2/MPP/java/lab5/lab5/Networking/src/main/java/ro/mpp2024/network/utils/RpcConcurrentServer.java
package ro.mpp2024.network.utils;

import ro.mpp2024.network.rpcprotocol.ClientRpcReflectionWorker;
import ro.mpp2024.services.IServices;

import java.net.Socket;

public class RpcConcurrentServer extends AbsConcurrentServer {
    private IServices Server;
    public RpcConcurrentServer(int port, IServices transportServer) {
        super(port);
        this.Server = transportServer;
        System.out.println(" RpcConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        ClientRpcReflectionWorker worker = new ClientRpcReflectionWorker(Server, client);

        Thread tw = new Thread(worker);
        return tw;
    }

    @Override
    public void stop(){
        System.out.println("Stopping services ...");
    }
}

File: AN_2/SEM_2/MPP/c#/lab6/Networking/mpp.networking/utils/RpcConcurrentServer.cs
using System.Globalization;
using Networking.mpp.networking.jsonprotocol;

namespace Networking.mpp.networking.utils;
using Services.mpp.services;
using Networking.mpp.networking.rpcprotocol;

public class RpcConcurrentServer : AbsConcurrentServer
{
    private readonly IServices _server;

    public RpcConcurrentServer(int port, IServices server) : base(port)
    {
        this._server = server;
        Console.WriteLine("RpcConcurrentServer started.");
    }

    protected override Thread CreateWorker(TcpClient client)
    {
        ClientJsonWorker worker = new ClientJsonWorker(_server, client);
        return new Thread(worker.Run);
    }

    public new void Stop()
    {
        Console.WriteLine("Stopping services...");
        base.Stop();
    }
}